pkg/daemon/cleanup/cleanup.go

```diff
@@ -65,13 +75,47 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 		return err
 	}
 	cleanupContainers()
-
+	if len(latestCMData.EnvironmentVars) != 0 {
```

Comment:
Shouldn't we use the CM tied to `VersionAnnotation`? We could use the latest one as backup like we do in  getServicesToRemove()